R/getTimeDivergence.R

```diff
@@ -70,33 +70,48 @@ getTimeDivergence <- function(se,
 
     # Manipulate each subobject
     se_more_list <- lapply(seq_along(spl_more),
-                function(i){.check_pairwise_dist(x = se[, spl_more[[i]]],
-                                                distfun=distfun,
-                                                time_interval,
-                                                name_divergence = "time_divergence",
-                                                name_timedifference = "time_difference",
-                                                time_field,
-                                                abund_values)})
+        function(i){.check_pairwise_dist(x = se[, spl_more[[i]]],
+                                        distfun=distfun,
+                                        time_interval,
+                                        name_divergence = "time_divergence",
+                                        name_timedifference = "time_difference",
+                                        time_field,
+                                        abund_values)})
 
-    se_one_list <- lapply(seq_along(spl_one), function(i) {
-        se[, spl_one[[i]]]}
-    )
+            se_one_list <- lapply(seq_along(spl_one), function(i) {
+                se[, spl_one[[i]]]}
+            )
 
-    # assign the names back to (T)SE objects
-    names(se_more_list) <- names(spl_more)
-    names(se_one_list) <- names(spl_one)
+            for(i in seq_along(se_one_list)){
+                colData(se_one_list[[i]])[, name_timedifference] <- NA
+                colData(se_one_list[[i]])[, name_divergence] <- NA
+            }
 
-    # put lists together and put them in order
-    whole_se <- do.call(c, list(se_one_list, se_more_list))
-    whole_se <- whole_se[order(as.numeric(names(whole_se)))]
+            # assign the names back to (T)SE objects
+            names(se_more_list) <- names(spl_more)
+            names(se_one_list) <- names(spl_one)
 
-    # Merge the objects back into a single SE
-    se_new <- mergeSEs(whole_se)
+            # put lists together and put them in order
+            whole_se <- do.call(c, list(se_one_list, se_more_list))
+            whole_se <- whole_se[order(as.numeric(names(whole_se)))]
 
-    return(se_new)
+            # Merge the objects back into a single SE
 
-}
+            whole_se<-whole_se[!sapply(whole_se,is.null)]
+
+    #  if (class(whole_se[[1]]) == "TreeSummarizedExperiment"){
+    #      data <- cbind(whole_se[[1]], whole_se[[2]])
+
+    #      for( i in 3:length(whole_se)){
+    #          data <- cbind(data, whole_se[[i]])
+    #      }
+    #     return(data)
+    #  }
```

Comment:
Sorry, I saw this PR pretty late. So what I found is that TreeSE binds using `cbind`. However, using this method returns the correct answer as well as error: 

unable to find an inherited method for function 'bindCOLS' for signature 'TreeSummarizedExperiment'.

Reply:
What's the action point and next steps? Or open questions?

Reply:
I found a way to fix this problem

Reply:
This problem is solved now.